src/gen/conditions.rs

```diff
@@ -767,6 +766,65 @@ fn cond_test(input: &str) -> Result<(Allocator, Vec<SpendConditionSummary>), Val
     cond_test_cb(input, None)
 }
 
+#[test]
+fn test_invalid_condition_list1() {
+    assert_eq!(
+        cond_test("((({h1} ({h2} (123 (8 )))").unwrap_err().1,
+        ErrorCode::InvalidCondition
+    );
+}
+
+#[test]
+fn test_invalid_condition_list2() {
+    assert_eq!(
+        cond_test("((({h1} ({h2} (123 ((8 ))))").unwrap_err().1,
+        ErrorCode::InvalidCondition
+    );
+}
+
+#[test]
+fn test_invalid_condition_list_terminator() {
```

Comment:
I think `test_invalid_condition_list_terminator` needs more comments.
The name of the test suggests to the reader that the test should fail, but we understand that it is just the list that is invalid, not the test case.

This is also a great place to refer to the equivalent test in 

https://github.com/Chia-Network/chia-blockchain/pull/8393/files#diff-9439f549ee25ef5e6c0c4986a8710a0d5bb2a18d8b3c492ef41edbeb4e48c88fR2064

where you use `generator_condition_tester("(80 50 . 1)", ...)`
Translating that syntax in a comment might help someone familiar with clvm syntax.